api/infrastructure/parallelcluster-api.yaml

```diff
@@ -258,8 +268,12 @@ Resources:
                - apigateway.amazonaws.com
            Action:
              - 'sts:AssumeRole'
+     PermissionsBoundary: !If [HasPermissionBoundary, !Ref PermissionsBoundaryPolicy, !Ref 'AWS::NoValue']
+     RoleName: !Sub
+       - ${IAMRoleAndPolicyPrefix}APIGatewayExecutionRole-${StackIdSuffix}
+       - { StackIdSuffix: !Select [ 0, !Split [ '-', !Select [ 2, !Split [ '/', !Ref 'AWS::StackId']]]] }
```

Comment:
for example if we have stack id suffix as the following: -7107dab0-3bdf-11ee-bca9-0e94062cc5d1, to fit in the role name, prefix and disambiguator, he suggested only using the first part of the stack id suffix so in this case it would be 7107dab0 as the stack id suffix instead of the entire thing